renderdoc/driver/vulkan/vk_replay.cpp

```diff
@@ -1490,32 +1496,53 @@ void VulkanReplay::SavePipelineState(uint32_t eventId)
     }
 
     // Color Blend
-    ret.colorBlend.alphaToCoverageEnable = p.alphaToCoverageEnable;
-    ret.colorBlend.alphaToOneEnable = p.alphaToOneEnable;
+    ret.colorBlend.alphaToCoverageEnable = state.alphaToCoverageEnable != VK_FALSE;
+    ret.colorBlend.alphaToOneEnable = state.alphaToOneEnable != VK_FALSE;
 
-    ret.colorBlend.blends.resize(p.attachments.size());
-    for(size_t i = 0; i < p.attachments.size(); i++)
+    // find size if no static pipeline state
+    size_t numAttach = std::max({state.colorBlendEnable.size(), state.colorBlendEquation.size(),
+                                 state.colorWriteEnable.size(), state.colorWriteMask.size()});
```

Comment:
Something I appreciate might feel nitpicky, generally I avoid `std::` where possible. This `std::max` overload might be supported as the docs seem to suggest it's C++11, but I would bet good money that Android will probably ruin that somehow as it always does - I don't know if the Android compilers we target fully support C++11 as I've had problems with that in the past.

I'd rather you use `RDCMAX` here and do it one by one for each value.

Reply:
This has been changed in the latest update.